packages/lesswrong/lib/collections/posts/schema.tsx

```diff
@@ -1200,6 +1200,35 @@ const schema: SchemaType<"Posts"> = {
     })
   }),
 
+  spotlight: resolverOnlyField({
+    type: "Spotlight",
+    graphQLtype: "Spotlight",
+    canRead: ['guests'],
+    resolver: async (post: DbPost, args: void, context: ResolverContext) => {
+      const { currentUser, Spotlights } = context;
+      const spotlight = await getWithLoader(context, Spotlights,
+        "postSpotlight",
+        {
+          documentId: post._id,
+          draft: false,
+          deletedDraft: false
+        },
+        "documentId", post._id
+      );
+      return accessFilterSingle(currentUser, Spotlights, spotlight[0], context);
+    },
+    sqlResolver: ({field, join}) => join({
+      table: "Spotlights",
+      type: "left",
+      on: {
+        documentId: field("_id"),
```

Comment:
This may break in fun and unexpected ways if there's ever more than one published spotlight with the same post id. Edit: just saw the mutation that should avoid this happening - maybe a warning comment would be sensible somewhere anyway.